docs/Homebrew-and-Python.md

```diff
@@ -27,7 +27,9 @@ $(brew --prefix python)/libexec/bin
 
 ## Setuptools, pip, etc.
 
-The Python formulae install [pip](https://pip.pypa.io/) (as `pip3`) and [Setuptools](https://pypi.org/project/setuptools/).
+The Python formulae install [pip](https://pip.pypa.io/) (as `pip3`). Python@3.11 and older Python formulae also install [Setuptools](https://pypi.org/project/setuptools/).
+
+Starting with Python@3.12, the bundled Python packages should be updated by reinstalling brewed Python. For older Python formulae, they can be updated as described below.
```

Comment:
Not sure of phrasing here, but `pip install --upgrade` will hit PEP 668, e.g.
```console
❯ python3 -m pip install --upgrade setuptools
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try brew install
    xyz, where xyz is the package you are trying to
    install.
```

Went with **should** rather than **must** since it is possible to override via `--break-system-packages`.

---

Also not sure the differences in our choice of name style, i.e.
- Python 3.11
- Python@3.11
- `python@3.11`